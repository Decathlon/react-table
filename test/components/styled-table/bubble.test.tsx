/// <reference path="../../typings/tests-entry.d.ts" />
import { createRenderer } from "react-test-renderer/shallow";

import Bubble, { BubbleType } from "../../../src/components/styled-table/bubble";

describe("Bubble component", () => {
  test("should render the default bubble", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Bubble badge="30" />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a bubble without badge", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Bubble className="foo-class-name" />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a bubble with content", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(
      <Bubble badge="30">
        <div>Foo</div>
      </Bubble>
    );
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a success bubble with content", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(
      <Bubble badge="30" type={BubbleType.success}>
        <div>Foo</div>
      </Bubble>
    );
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});
