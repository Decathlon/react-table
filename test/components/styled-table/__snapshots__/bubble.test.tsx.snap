// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bubble component should render a bubble with content 1`] = `
<div
  className="bubble-container"
>
  <div>
    Foo
  </div>
  <svg
    className="bubble-circle"
    height="100%"
    viewBox="0 0 100 100"
    width="100%"
  >
    <circle
      className="bubble-circle-main"
      cx="50"
      cy="50"
      r="45"
    />
    <React.Fragment>
      <circle
        className="bubble-circle-content"
        cx="80"
        cy="20"
        r="10"
      />
      <text
        className="bubble-circle-text-content"
        x={80}
        y={24}
      >
        30
      </text>
    </React.Fragment>
  </svg>
</div>
`;

exports[`Bubble component should render a bubble without badge 1`] = `
<div
  className="bubble-container"
>
  <svg
    className="bubble-circle foo-class-name"
    height="100%"
    viewBox="0 0 100 100"
    width="100%"
  >
    <circle
      className="bubble-circle-main"
      cx="50"
      cy="50"
      r="45"
    />
  </svg>
</div>
`;

exports[`Bubble component should render a success bubble with content 1`] = `
<div
  className="bubble-container"
>
  <div>
    Foo
  </div>
  <svg
    className="bubble-circle success"
    height="100%"
    viewBox="0 0 100 100"
    width="100%"
  >
    <circle
      className="bubble-circle-main"
      cx="50"
      cy="50"
      r="45"
    />
    <React.Fragment>
      <circle
        className="bubble-circle-content"
        cx="80"
        cy="20"
        r="10"
      />
      <text
        className="bubble-circle-text-content"
        x={80}
        y={24}
      >
        30
      </text>
    </React.Fragment>
  </svg>
</div>
`;

exports[`Bubble component should render the default bubble 1`] = `
<div
  className="bubble-container"
>
  <svg
    className="bubble-circle"
    height="100%"
    viewBox="0 0 100 100"
    width="100%"
  >
    <circle
      className="bubble-circle-main"
      cx="50"
      cy="50"
      r="45"
    />
    <React.Fragment>
      <circle
        className="bubble-circle-content"
        cx="80"
        cy="20"
        r="10"
      />
      <text
        className="bubble-circle-text-content"
        x={80}
        y={24}
      >
        30
      </text>
    </React.Fragment>
  </svg>
</div>
`;
